src/v/debug_bundle/BUILD

```diff
@@ -0,0 +1,30 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+
+redpanda_cc_library(
+    name = "debug_bundle",
+    srcs = [
+        "debug_bundle_service.cc",
+        "error.cc",
+        "types.cc",
+    ],
+    hdrs = [
+        "debug_bundle_service.h",
+        "error.h",
+        "fwd.h",
+        "types.h",
+    ],
+    include_prefix = "debug_bundle",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+        "//src/v/config",
+        "//src/v/container:fragmented_vector",
```

Comment:
Is this and other deps like security needed? Can we remove to be the minimal set here please?

Also if there are deps that are included in implementation files only, you can move them to implementation_deps which doesn't propagate them transitively

Reply:
Yes, they're used in the `types.h` header which is a public header (see the `hdrs` list).  Am I missing something?  I thought `deps` was the list of public deps required by users of this library and `internal_deps` is just required to build the library.